### ISSUE TITLE
Podman: ERROR: UserNS: memory controller needs to be delegated

### ISSUE THREAD

**Issue by a1exus** [kubernetes/minikube#18692]
### What Happened?

can't start a local kubernetes cluster
```
toor@suey:~$ minikube version
minikube version: v1.32.0
commit: 8220a6eb95f0a4d75f7f2d7b14cef975f050512d
toor@suey:~$ podman info 
host:
  arch: amd64
  buildahVersion: 1.33.5
  cgroupControllers:
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon_2.1.10+ds1-1_amd64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.10, commit: unknown'
  cpuUtilization:
    idlePercent: 96.02
    systemPercent: 1.19
    userPercent: 2.79
  cpus: 8
  databaseBackend: boltdb
  distribution:
    codename: kali-rolling
    distribution: kali
    version: "2024.1"
  eventLogger: journald
  freeLocks: 2047
  hostname: suey.nknwn.local
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.15.0-kali2-rt-amd64
  linkmode: dynamic
  logDriver: journald
  memFree: 24614518784
  memTotal: 33545621504
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns_1.4.0-5_amd64
      path: /usr/lib/podman/aardvark-dns
      version: aardvark-dns 1.4.0
    package: netavark_1.4.0-4_amd64
    path: /usr/lib/podman/netavark
    version: netavark 1.4.0
  ociRuntime:
    name: crun
    package: crun_1.14.4-1_amd64
    path: /usr/bin/crun
    version: |-
      crun version 1.14.4
      commit: a220ca661ce078f2c37b38c92e66cf66c012d9c1
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt_0.0~git20240326.4988e2b-1_amd64
    version: |
      pasta 0.0~git20240326.4988e2b-1
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns_1.2.1-1_amd64
    version: |-
      slirp4netns version 1.2.1
      commit: 09e31e92fa3d2a1d3ca261adaeb012c8d75a8194
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.5
  swapFree: 0
  swapTotal: 0
  uptime: 1h 10m 29.00s (Approximately 0.04 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries: {}
store:
  configFile: /home/toor/.config/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/toor/.local/share/containers/storage
  graphRootAllocated: 210579116032
  graphRootUsed: 79115735040
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Supports shifting: "true"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 2
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/toor/.local/share/containers/storage/volumes
version:
  APIVersion: 4.9.3
  Built: 0
  BuiltTime: Wed Dec 31 19:00:00 1969
  GitCommit: ""
  GoVersion: go1.21.6
  Os: linux
  OsArch: linux/amd64
  Version: 4.9.3

toor@suey:~$ MINIKUBE_ROOTLESS=true minikube start --driver podman --alsologtostderr
...
-- /stdout --
I0418 21:51:57.875844   27327 cli_runner.go:164] Run: podman logs --timestamps minikube
- I0418 21:51:57.994270   27327 errors.go:91] Postmortem logs ("podman logs --timestamps minikube"): -- stdout --
2024-04-18T21:51:35.219031000-04:00 + userns=
2024-04-18T21:51:35.219297000-04:00 + grep -Eqv '0[[:space:]]+0[[:space:]]+4294967295' /proc/self/uid_map
2024-04-18T21:51:35.225507000-04:00 + userns=1
2024-04-18T21:51:35.225658000-04:00 + echo 'INFO: running in a user namespace (experimental)'
2024-04-18T21:51:35.225768000-04:00 INFO: running in a user namespace (experimental)
2024-04-18T21:51:35.227145000-04:00 + validate_userns
2024-04-18T21:51:35.227257000-04:00 + [[ -z 1 ]]
2024-04-18T21:51:35.227331000-04:00 + local nofile_hard
2024-04-18T21:51:35.227536000-04:00 ++ ulimit -Hn
2024-04-18T21:51:35.227742000-04:00 + nofile_hard=1048576
2024-04-18T21:51:35.227812000-04:00 + local nofile_hard_expected=64000
2024-04-18T21:51:35.227880000-04:00 + [[ 1048576 -lt 64000 ]]
2024-04-18T21:51:35.227947000-04:00 + [[ -f /sys/fs/cgroup/cgroup.controllers ]]
2024-04-18T21:51:35.228009000-04:00 + for f in cpu memory pids
2024-04-18T21:51:35.228083000-04:00 + grep -qw cpu /sys/fs/cgroup/cgroup.controllers
2024-04-18T21:51:35.229288000-04:00 + for f in cpu memory pids
2024-04-18T21:51:35.229389000-04:00 + grep -qw memory /sys/fs/cgroup/cgroup.controllers
2024-04-18T21:51:35.230796000-04:00 + echo 'ERROR: UserNS: memory controller needs to be delegated'
2024-04-18T21:51:35.230900000-04:00 ERROR: UserNS: memory controller needs to be delegated
2024-04-18T21:51:35.230972000-04:00 + exit 1
```

### Attach the log file

```
toor@suey:~$ minikube logs --file=log.txt

❌  Exiting due to GUEST_STATUS: Unable to get machine status: state: unknown state "minikube": podman container inspect minikube --format=<no value>: exit status 125
stdout:

stderr:
Error: no such container minikube


╭───────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                                           │
│    😿  If the above advice does not help, please let us know:                             │
│    👉  https://github.com/kubernetes/minikube/issues/new/choose                           │
│                                                                                           │
│    Please run `minikube logs --file=logs.txt` and attach logs.txt to the GitHub issue.    │
│    Please also attach the following file to the GitHub issue:                             │
│    - /tmp/minikube_logs_747bd9a87b9ef40c507ef76c9306a0d0af3c8b26_0.log                    │
│                                                                                           │
╰───────────────────────────────────────────────────────────────────────────────────────────╯

toor@suey:~$ 
```

### Operating System

Other

### Driver

Podman

**Comment by a1exus** [comment]
my env already have this:
```
toor@suey:~$ cat /etc/systemd/system/user@.service.d/delegate.conf 
[Service]
Delegate=cpu cpuset io memory pids
toor@suey:~$ sudo systemctl daemon-reload
toor@suey:~$ echo $?
0
toor@suey:~$ 
```

**Comment by a1exus** [comment]
i've fixed it in my env by upgrading my linux kernel from 5.15.5 to 6.6.15, probably upgrades of cgroups maybe parts of systemd..

```
toor@suey:~$ MINIKUBE_ROOTLESS=true minikube start --driver podman
😄  minikube v1.32.0 on Debian kali-rolling
    ▪ MINIKUBE_ROOTLESS=true
✨  Using the podman driver based on user configuration
📌  Using rootless Podman driver
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
E0425 14:12:13.225974    9643 cache.go:189] Error downloading kic artifacts:  not yet implemented, see issue #8426
🔥  Creating podman container (CPUs=2, Memory=7900MB) ...
🐳  Preparing Kubernetes v1.28.3 on Docker 24.0.7 ...
❗  Certificate client.crt has expired. Generating a new one...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
toor@suey:~$ 
```
